test-utils/src/main/groovy/io/micronaut/kubernetes/test/KubernetesSpecification.groovy

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.test
+
+import groovy.util.logging.Slf4j
+import io.fabric8.kubernetes.api.model.IntOrString
+import io.fabric8.kubernetes.api.model.ServicePortBuilder
+import io.fabric8.kubernetes.api.model.ServiceSpecBuilder
+import io.fabric8.kubernetes.api.model.apps.Deployment
+import io.micronaut.context.annotation.Value
+import io.micronaut.core.io.ResourceResolver
+import io.micronaut.core.io.scan.ClassPathResourceLoader
+import spock.lang.AutoCleanup
+import spock.lang.Shared
+import spock.lang.Specification
+
+import javax.inject.Inject
+
+/**
+ * Abstract specification encapsulating setup of test namespace.
+ *
+ * @author Pavol Gressa
+ * @since 2.3
+ */
+@Slf4j
+abstract class KubernetesSpecification extends Specification {
```

Comment:
This is basically the main class that does the setup of resources that was previously in the shell file.